quickboard/base/_datamanager.py

```diff
@@ -0,0 +1,75 @@
+import pandas as pd
+
+from dash.dependencies import Input, Output, State, ALL
+
+from quickboard.dashsetup import app
+
+
+class DataManager:
+    """
+    A class for interpreting different data sources, and cleaning data into a Pandas Dataframe for use
+    by DynamicPanel objects. The possible types of data_source are:
+        - pandas DataFrame (loaded in memory)
+        - file path ending in .csv or .tsv (to be loaded into a DataFrame)
+        - a list with first element a PlotPanel and second element a string with value either hoverData, clickData, or
+        selectedData to be used for data generated from interacting with given PlotPanel.
+    """
+    def __init__(self, data_source=""):
+        self.data_source = data_source
+        self.source_type = None
+        self.sub_df = pd.DataFrame()  # df after sidebar transforms
+
+        if isinstance(data_source, pd.DataFrame):
+            self.source_type = "DataFrame"
+
+        elif isinstance(data_source, str):
+            extension = data_source.split('.')[-1]
+            if extension == 'tsv':
+                self.source_type = "tsv"
+            elif extension == 'csv':
+                self.source_type = "csv"
+            else:
+                self.source_type = "tab"  # All other strings interpreted as use tab data
+
+        elif isinstance(data_source, list):
+            # Must have first entry a PlotPanel, and second entry string (one of: hoverData, clickData, selectedData)
+            self.source_type = "PlotPanel"
+
+        else:
+            self.df = pd.DataFrame()
+            print("ERROR: Invalid data_source input. Please see documentation for list of valid input types.")
```

Comment:
Could you throw an actual error here now rather than just printing a message?

Reply:
Should be fixed now